src/js/panels.js

```diff
@@ -16,20 +15,110 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+        },
+        invokers: {
+            stepUp: {
+                funcName: "gpii.firstDiscovery.panel.ranged.step",
+                args: ["{that}"]
+            },
+            stepDown: {
+                funcName: "gpii.firstDiscovery.panel.ranged.step",
+                args: ["{that}", true]
+            },
+            updateMeter: {
+                funcName: "gpii.firstDiscovery.panel.ranged.updateMeter",
+                args: ["{that}", "{that}.model.value"]
+            }
+        },
+        listeners: {
+            "afterRender.bindIncrease": {
+                "this": "{that}.dom.increase",
+                "method": "click",
+                "args": ["{that}.stepUp"]
+            },
+            "afterRender.bindDecrease": {
+                "this": "{that}.dom.decrease",
+                "method": "click",
+                "args": ["{that}.stepDown"]
+            },
+            "afterRender.updateMeter": "{that}.updateMeter"
+        },
+        modelListeners: {
+            "value": {
+                listener: "{that}.updateMeter",
+                excludeSource: ["init"]
             }
         }
     });
 
+    gpii.firstDiscovery.panel.ranged.clip = function (value, min, max) {
+        if (max > min) {
+            return Math.min(max, Math.max(min, value));
+        }
+    };
+
+    gpii.firstDiscovery.panel.ranged.step = function (that, reverse) {
+        var step = reverse ? (that.options.step * -1) : that.options.step;
+        var newValue = that.model.value + step;
+        newValue = gpii.firstDiscovery.panel.ranged.clip(newValue, that.options.range.min, that.options.range.max);
+        that.applier.change("value", newValue);
+    };
+
+    gpii.firstDiscovery.panel.ranged.calculatePercentage = function (value, min, max) {
+        if (max > min) {
+            var clipped = gpii.firstDiscovery.panel.ranged.clip(value, min, max);
+            return ((clipped - min) / (max - min)) * 100;
+        }
+    };
+
+    gpii.firstDiscovery.panel.ranged.updateMeter = function (that, value) {
+        var percentage = gpii.firstDiscovery.panel.ranged.calculatePercentage(value, that.options.range.min, that.options.range.max);
+        that.locate("meter").css("height", percentage + "%");
+    };
+
     fluid.defaults("gpii.firstDiscovery.panel.textSize", {
-        gradeNames: ["fluid.prefs.panel", "autoInit"],
+        gradeNames: ["gpii.firstDiscovery.panel.ranged", "autoInit"],
         preferenceMap: {
             "fluid.prefs.textSize": {
-                "model.textSize": "default",
+                "model.value": "default",
+                "range.min": "minimum",
+                "range.max": "maximum",
+                "step": "divisibleBy"
+            }
+        }
+    });
+
+    fluid.defaults("gpii.firstDiscovery.panel.audio", {
+        gradeNames: ["fluid.prefs.panel", "autoInit"],
+        preferenceMap: {
+            "gpii.firstDiscovery.audio": {
```

Comment:
Shall we use "fluid.prefs.speak" here to match up with what's used in the schema?